examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,216 @@
+# for making allocation decisions under uncertainty.
+
+for clf, name in clf_list:
+    clf.fit(X_train, y_train)
+    y_pred = clf.predict(X_test)
+
+    print(f"{name}\n"
+          f"\tBrier score: {viz_objects[name].brier_value}\n"
+          f"\tPrecision: {precision_score(y_test, y_pred)}\n"
+          f"\tRecall: {recall_score(y_test, y_pred)}\n"
+          f"\tF1: {f1_score(y_test, y_pred)}\n")
+
+# %%
+# Next, we will compare:
+#
+# * :class:`~sklearn.linear_model.LogisticRegression` (baseline)
+# * :class:`~sklearn.svm.LinearSVC` (:term:`decision_function` scores
+#   min-max scaled to [0,1], since they are not proper probabilities)
+# * :class:`~sklearn.svm.LinearSVC` with isotonic and sigmoid
+#   calibration (see :ref:`User Guide <calibration>`)
+#
+
+lr = LogisticRegression(C=1.)
+svc = LinearSVC(max_iter=10000)
+svc_isotonic = CalibratedClassifierCV(svc, cv=2, method='isotonic')
+svc_sigmoid = CalibratedClassifierCV(svc, cv=2, method='sigmoid')
+
+clf_list = [(lr, 'Logistic'),
+            (svc, 'SVC'),
+            (svc_isotonic, 'SVC + Isotonic'),
+            (svc_sigmoid, 'SVC + Sigmoid')]
+
+fig = plt.figure(figsize=(10, 10))
+ax1 = plt.subplot2grid((4, 2), (0, 0), rowspan=2, colspan=2)
+
+viz_objects = {}
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    viz = plot_calibration_curve(
+        clf, X_test, y_test, n_bins=10, name=name, ax=ax1, color=colors(i)
+    )
+    viz_objects[name] = viz
+
+ax1.grid()
+ax1.set_title('Calibration plots (SVC)')
+ax1.set(xlabel="")
+
+# Add histogram
+for i, (_, name) in enumerate(clf_list):
+    if i <= 1:
+        ax = plt.subplot2grid((4, 2), (2, i))
+    else:
+        ax = plt.subplot2grid((4, 2), (3, i - 2))
+
+    ax.hist(
+        viz_objects[name].y_prob, range=(0, 1), bins=10, label=name,
+        color=colors(i)
+    )
+    ax.set(title=name, xlabel="Mean predicted probability", ylabel="Count")
+
+plt.tight_layout()
 plt.show()
+
+# %%
+# :class:`~sklearn.svm.LinearSVC` shows the opposite
+# behavior to :class:`~sklearn.naive_bayes.GaussianNB`; the calibration
+# curve has a sigmoid shape, which is typical for an under-confident
+# classifier. In the case of :class:`~sklearn.svm.LinearSVC`, this is caused
+# by the margin property of the hinge loss, which lets the model focus on
+# difficult samples that are close to the decision boundary (the support
+# vectors).
+#
+# Both kinds of calibration (sigmoid and isotonic) can fix this issue and
+# yield nearly identical results.
+#
+# Below, we print the Brier score, precision, recall and F1 score again.
+
+for clf, name in clf_list:
+    clf.fit(X_train, y_train)
+    y_pred = clf.predict(X_test)
+
+    print(f"{name}\n"
+          f"\tBrier Score: {viz_objects[name].brier_value}\n"
+          f"\tPrecision: {precision_score(y_test, y_pred)}\n"
+          f"\tRecall: {recall_score(y_test, y_pred)}\n"
+          f"\tF1: {f1_score(y_test, y_pred)}\n")
```

Comment:
Would this be nicer if we place this information in a pandas dataframe and display it?